### ISSUE TITLE
CLI: stack prompt multiline input when pasting

### ISSUE THREAD

**Issue by matti** [kontena/kontena#2838]
copy'n'paste, looks like this (but works)

![lol](https://user-images.githubusercontent.com/3893/30580686-19a13fb8-9d26-11e7-83c3-bf12593df446.png)

**Comment by kke** [comment]
This could be fixed in tty-prompt or by introducing some kind of "raw" prompt alternative for long strings in cli using `Readline.readline`:

```ruby
require 'readline'
Readline.pre_input_hook = proc { Readline.insert_text("abcd efg hijkl"); Readline.redisplay }
str = Readline.readline('> ')
```

It could look at the length of the `default:` value or `min_length:` and decide to use the raw prompt when it's high enough.

**Comment by piotrmurach** [comment]
Looking at the issue I'm wondering if asking for config data in such form is appropriate? Bar the obvious issue with `tty-prompt` I think each of these configuration options probably could do with separate `ask` prompt. The advantage would be to provide validations, shorter defaults and more flexibility. What if the defaults are all fine apart from one, let's say host value, would the user still have to copy&paste full config? In terms of asking for hash object, you can use [collect](https://github.com/piotrmurach/tty-prompt#28-collect) prompt that allows for nested keys and seems to match your needs quite well. 

I won't go the readline route as it doesn't work on Windows and provides huge dependency that `tty-prompt` doesn't need. In terms of auto-detecting the reading mode I think that's also a problematic solution. I would rather have a separate prompt just for this occasion. I will investigate long strings input but it won't be any time soon as I'm preparing for a conference.

**Comment by matti** [comment]
@piotrmurach what about using readline on !windows ?

**Comment by piotrmurach** [comment]
@matti I won't be using readline as this is a pure unix implementation with archaic api that's not needed in `tty-prompt`. Bringing huge dependency for this is something I'm not really prepared to do at this stage. The input reading is done by [tty-reader](https://github.com/piotrmurach/tty-reader) and I would be keen to make sure that it can  read the long input before suggesting any other solutions. Therefore I would start with `tty-reader` and making sure it can handle long input. As you have stated yourself it works but display is mingled which makes me think that it has nothing to do with input reading at all but how the prompt handles long strings and wrapping of content. In other words, it feels to me that readline is a solution to non-existing problem. Without investigating more I cannot say what is the real issue but what you have demonstrated may not be related to actual characters reading at all but more to how the prompt is displayed and how clearing of prompt is done through ANSI codes.

**Comment by kke** [comment]
I suppose there could be some sort of buffer in `prompt.ask` that would start echoing back only if `time_since_last_char > x_nanoseconds`, but anyways, this is not the right place for that discussion :) 

Readline is already used in kontena-cli if you install the console plugin (which works on windows using readline-rb). That idea was mainly for working around the problem outside of tty-prompt until the issue perhaps gets fixed one day. It just looks annoying but still works, so it's not that big of an issue.

**Comment by piotrmurach** [comment]
This is fixed in `v0.14.0` of `tty-prompt` and allows for pasting not only long strings but also content that wraps terminal width as well.
